lib/vehicle.rb

```diff
@@ -6,20 +6,37 @@ class Vehicle
               :make,
               :model,
               :engine
+  attr_accessor :registration_date, :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = nil
+    @plate_type = nil
   end
 
   def antique?
     Date.today.year - @year > 25
   end
 
+  def regular?
+    @engine == :ice
+  end
+
   def electric_vehicle?
     @engine == :ev
   end
+
+  def plate_type!
```

Comment:
I don't see a test in the vehicle_spec.rb for this method. Remember that every method should be tested in the test for the class.